### PR TITLE
[LIVE-10137] [CI] - [Github Actions] - Improve clarity of run actions

### PR DESCRIPTION
**Pull request by jdabbech-ledger**
### 📝 Description
Improve clarity of git run actions in workflows

### ❓ Context

- **JIRA or GitHub link**: [LIVE-10137](https://ledgerhq.atlassian.net/browse/LIVE-10137) 

### ✅ Checklist

Pull Requests must pass the CI and be code reviewed. Set as Draft if the PR is not ready.

- [ ] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** 
- [ ] **Impact of the changes:** 
  - 

---

### 🧐 Checklist for the PR Reviewers

- [ ] **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- [ ] **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- [ ] **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- [ ] **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- [ ] **Any new dependencies** have been justified and documented.
- [ ] **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[LIVE-10137]: https://ledgerhq.atlassian.net/browse/LIVE-10137?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/ledger-live-mobile/scripts/e2e-ci.sh [hunk 1]
```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env bash
```
Comment (aussedatlo) [inline apps/ledger-live-mobile/scripts/e2e-ci.sh:1]: You need to add `set -e` at the beginning of the file.

Right now, if one of the command inside the build/test function encounters an error, the script will continue and will return 0. So it's impossible to catch an error from the CI side.

With `set -e` if a command encounters an error, the script will halt and will return the matching error code.

File: apps/ledger-live-mobile/scripts/e2e-ci.sh [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+build_ios() {
+  pnpm mobile bundle:ios --dev false --minify false
+  pnpm mobile exec detox clean-framework-cache
+  pnpm mobile exec detox build-framework-cache
+  pushd apps/ledger-live-mobile
+  cp main.jsbundle ios/build/Build/Products/Release-iphonesimulator/ledgerlivemobile.app/main.jsbundle
+  mv main.jsbundle ios/build/Build/Products/Release-iphonesimulator/main.jsbundle
+  exit 0
```
Comment (aussedatlo) [inline apps/ledger-live-mobile/scripts/e2e-ci.sh:19]: no need to `exit 0` here

File: apps/ledger-live-mobile/scripts/e2e-ci.sh [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  usage
+fi
+
+if [ "$BUILD" ]; then
+  build_$PLATFORM
+fi
+
+if [ "$TEST" ]; then
+  test_$PLATFORM
+fi
```
Comment (aussedatlo) [inline apps/ledger-live-mobile/scripts/e2e-ci.sh:85]: no new line after the `fi` ?

File: apps/ledger-live-desktop/scripts/ci-lint.mjs [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+      --filter="ledger-live-desktop" \\
+      --api="http://127.0.0.1:${port}" \\
+      --token="yolo" \\
+      --team="foo" \\
+      -- --format="json" \\
+      -o="lint.json"`;
+  } else {
+    await $`pnpm lint \\
+      --filter="ledger-live-desktop" \\
+      --token="yolo" \\
```
Comment (valpinkman) [inline apps/ledger-live-desktop/scripts/ci-lint.mjs:42]: sorry I have added this as a parameter, we need to pass the server token (yolo) as argument

File: apps/ledger-live-desktop/scripts/ci-lint.mjs [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+const lint = async () => {
+  cd('../../')
+
+  if (typeof port === 'string') {
+    await $`pnpm lint \\
+      --filter="ledger-live-desktop" \\
+      --api="http://127.0.0.1:${port}" \\
+      --token="yolo" \\
```
Comment (valpinkman) [inline apps/ledger-live-desktop/scripts/ci-lint.mjs:35]: same here, we need to pass the token as arg

**Review by valpinkman — CHANGES_REQUESTED** [review]
Except for the token that needs to be an argument now, everything seems fine to me ;)

**Review by valpinkman — CHANGES_REQUESTED** [review]

**Review by valpinkman — APPROVED** [review]

**Pull request merged**